six/two/two.cpp

```diff
@@ -553,9 +551,28 @@ std::string Two::_fetchPythonError()
             if (format_exception != NULL) {
                 PyObject *fmt_exc = PyObject_CallFunctionObjArgs(format_exception, ptype, pvalue, ptraceback, NULL);
                 if (fmt_exc != NULL) {
+                    // Unlikely to happen but PyList_Size might fail, handle error
+                    Py_ssize_t len = PyList_Size(fmt_exc);
+                    if (len == -1) {
+                        // don't fetch the actual error or the caller might think
+                        // it was the root cause, while it's not. Just return
+                        // "unknown error".
+                        PyErr_Clear();
```

Comment:
I think `PyErr_Clear()` is idempotent, so no harm done, but we call `PyErr_Clear()` in `done:`